### PR TITLE
Fix README typos and improved syntax a bit

### PR DESCRIPTION
**Pull request by syrusakbary**
While reading the README I realized a few typos (nubmers, langauge, Markle) so I took the liberty of running the README through Grammarly to spot more errors and hence this PR!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -426,33 +425,33 @@ first place.
...
    .a file extension and named after its contents. For example, the
    archive file containing all libc objects is named `libc.a`.
 
    If you pass an archive file along with other object files to the
    linker, the linker pulls out an object file from the archive _only
    when_ it is referenced by other object files. In other words,
    unlike object files directly given to a linker, object files
-   wrapped in an archive are not linked to an output by default.
-   An archive works as supplements to complete your program.
+   wrapped in an archive are not linked to output by default.
```
Comment (Alcaro) [inline README.md:436]: Should probably be 'the output'.

**Review by Alcaro — COMMENTED** [review]
Other than those three, this looks like a significant improvement.

File: README.md [hunk 2]
```diff
@@ -294,10 +293,10 @@ tool.
   an output file. This is a slow step, but we can speed it up by
   splitting a file into small chunks, computing SHA-1 for each chunk,
   and then computing SHA-1 of the concatenated SHA-1 hashes
-  (i.e. constructing a [Markle
+  (i.e. constructing a [Merkle
   Tree](https://en.wikipedia.org/wiki/Merkle_tree) of height 2).
   Modern x86 processors have purpose-built instructions for SHA-1 and
-  can compute SHA-1 pretty quickly at about 2 GiB/s rate. Using 16
+  can compute SHA-1 pretty quickly at about 2 GiB/s rates. Using 16
```
[the remark this answers is not in the document]
Comment (syrusakbary) [reply README.md:299]:
```suggestion
  can compute SHA-1 pretty quickly at about 2 GiB/s. Using 16
```

File: README.md [hunk 3]
```diff
@@ -372,14 +371,14 @@ essential and which is not.
...
 a Unix linker has a coherent feature set for the Unix of that age. So,
-let me entangle the history to see how the operating system, runtime
+let me entangle the history to see how the operating system, runtime,
 and linker have gained features that we see today. That should give
-you an idea why a particular feature has been added to a linker in the
+you an idea of why a particular feature has been added to a linker in the
 first place.
 
-1. Original Unix didn't support shared library, and a program was
+1. Original Unix didn't support a shared library, and a program was
```
[the remark this answers is not in the document]
Comment (syrusakbary) [reply README.md:379]:
```suggestion
1. Original Unix didn't support shared libraries, and a program was
```

File: README.md (hunk 1 above)
Comment (syrusakbary) [reply README.md:436]:
```suggestion
   wrapped in an archive are not linked to the output by default.
```

File: README.md [hunk 4]
```diff
@@ -96,8 +96,7 @@ It is sometimes very hard to pass an appropriate command line
 option to `cc` to specify an alternative linker.
 To deal with the situation, mold has a feature to intercept all
 invocations of `/usr/bin/ld`, `/usr/bin/ld.lld` or `/usr/bin/ld.gold`
- and redirect it to itself. To use the feature, run `make` (or other
-build command) as a subcommand of mold as follows:
+ and redirect it to itself. To use the feature, run `make` (or another build command) as a subcommand of mold as follows:
```
Comment (syrusakbary) [inline README.md:99]:
```suggestion
and redirect it to itself. To use the feature, run `make` (or another
build command) as a subcommand of mold as follows:
````

**Comment by rui314** [comment]
Thanks! These commits have been merged.

**Pull request merged**